### ISSUE TITLE
Resizing the LED number causes flickering until ESP32 reboot.

### ISSUE THREAD

**Issue by sblantipodi** [Makuna/NeoPixelBus#520]
**Describe the bug**
Resizing the LED number causes flickering until ESP32 reboot.

**To Reproduce**
Inizialize a LED strip with 500LEDs.
Reside the LED strip to 100LEDs.
Drive the LED strip with some effects and see the flickering.

**Expected behavior**
Reducing the LED number of the LED strip should not cause flickering.

**Development environment (please complete the following information):**
 - OS: Win10
 - Build Environment PlatformIO 5.2.1
 - Board target ESP32
 - Library version [e.g. v2.3.4] - latest 2.6.8

**Minimal Sketch that reproduced the problem:**
```
NeoPixelBus<NeoGrbFeature, NeoEsp32I2s1800KbpsMethod>* ledsESP32 = NULL; 
ledsESP32 = new NeoPixelBus<NeoGrbFeature, NeoEsp32I2s1800KbpsMethod>(500, 15); 
ledsESP32->Begin();
ledsESP32->Show();

if (ledsESP32 != NULL) {
    delete ledsESP32;
}

// change PIN number and LED number
ledsESP32 = new NeoPixelBus<NeoGrbFeature, NeoEsp32I2s1800KbpsMethod>(100, 5); // and recreate with new count
ledsESP32->Begin();
ledsESP32->Show();
```

This simple code creates flickering.

Basically I initialize a LED strip with 500LEDs on GPIO 15,
then clear it and init 100LEDs on GPIO 5.

In this way the strip flickers until ESP reboot.
